django/db/migrations/loader.py

```diff
@@ -97,17 +97,21 @@ def load_disk(self):
                 if was_loaded:
                     reload(module)
             self.migrated_apps.add(app_config.label)
-            directory = os.path.dirname(module.__file__)
-            # Scan for .py files
-            migration_names = set()
-            for name in os.listdir(directory):
-                if name.endswith(".py"):
-                    import_name = name.rsplit(".", 1)[0]
-                    if import_name[0] not in "_.~":
-                        migration_names.add(import_name)
+            migration_names = set(name for _, name, is_pkg in pkgutil.iter_modules(module.__path__) if not is_pkg)
             # Load them
             for migration_name in migration_names:
-                migration_module = import_module("%s.%s" % (module_name, migration_name))
+                try:
+                    migration_module = import_module("%s.%s" % (module_name, migration_name))
+                except ImportError as e:
+                    if 'bad magic number' in str(e):
+                        raise ImportError(
+                            "Couldn't import \"{0}.{1}\". Please make sure any stale .pyc files have been deleted. "
+                            "Hint: Try running 'find {2} -name \\*.pyc -delete'".format(
```

Comment:
I'm not sure about including the find command here and in the release notes. What about Windows users? :-)
I think it's easy enough to search "delete pyc files" and find a solution.